packages/react-instantsearch-hooks-dom/src/widgets/ToggleRefinement.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { useToggleRefinement } from 'react-instantsearch-hooks';
+
+import { ToggleRefinement as ToggleRefinementUiComponent } from '../ui/ToggleRefinement';
+
+import type { ToggleRefinementProps as ToggleRefinementUiComponentProps } from '../ui/ToggleRefinement';
+import type { UseToggleRefinementProps } from 'react-instantsearch-hooks';
+
+export type ToggleRefinementProps = Omit<
+  ToggleRefinementUiComponentProps,
+  'onChange' | 'checked'
+> &
+  UseToggleRefinementProps;
+
+export function ToggleRefinement({
+  attribute,
+  on,
+  off,
+  ...props
+}: ToggleRefinementProps) {
+  const { refine, value } = useToggleRefinement(
+    { attribute, on, off },
+    {
+      $$widgetType: 'ais.toggleRefinement',
+    }
+  );
+
+  return (
+    <ToggleRefinementUiComponent
+      {...props}
```

Comment:
```suggestion
      label={attribute}
      {...props}
```

In InstantSearch.js we have the equivalent as this, meaning the label isn't required. Not sure what's best though

Reply:
The problem is that `attribute` is the name of the record property, but you may need to specify a display name. In InstantSearch.js it's solved with the [`labelText`](https://www.algolia.com/doc/api-reference/widgets/toggle-refinement/js/#templates) template, but since we're not opening translations yet, it would mean users can't customize the display name.

Reply:
you would be able to, as props is still spread right after it, which would override the label, if given, this is just giving a default value to the label I'm suggesting :)

Reply:
Good point! Addressed in 0b96386b.